src/create-pr.js

```diff
@@ -0,0 +1,271 @@
+    const pr = await createReleasePR(getReleasePRBody(filteredPullRequests));
+    await addLabelToThePR(pr.number);
+
+    console.debug(`Pull Request Created: ${pr.html_url}`);
+  } catch (error) {
+    core.setFailed(error.message);
+  }
+}
+
+/**
+ * Add Release Label to the RC PR
+ *
+ * @param {string} prNumber pull request number
+ */
+async function addLabelToThePR (prNumber) {
+  await octokit.issues.update({
+    owner,
+    repo,
+    issue_number: prNumber,
+    labels: [RELEASE_PR_IDENTIFIER_LABEL]
+  });
+}
+
+/**
+ * Clone the repository
+ */
+async function cloneRepo () {
+  const REPO_URL = 'https://' + owner + ':' + GITHUB_TOKEN + '@github.com/' + owner + '/' + repo;
+
+  await simpleGit(GITHUB_WORKSPACE)
+    .clone(REPO_URL);
+}
+
+/**
+ * Create Release Candidate Branch with a new Commit
+ */
+async function createReleaseCandidateBranch () {
+  const cwd = GITHUB_WORKSPACE + '/' + repo;
+
+  _.each(REPLACE_COMMANDS, function (cmd) {
+    cmd = cmd.replace('<<DATE>>', dayjs().format('YYYY-MM-DD'));
+    execSync(cmd, { cwd });
+  });
+
+  await simpleGit(cwd)
+    .addConfig('user.name', COMMIT_USERNAME)
+    .addConfig('user.email', COMMIT_USEREMAIL)
+    .checkoutLocalBranch(getRcBranchName())
+    .add('./*')
+    .commit(FILE_UPDATE_COMMIT_DESC)
+    .push('origin', getRcBranchName())
+}
+
+/**
+ * Create Release Pull Request
+ *
+ * @param {string} body body of the pr to be created
+ * @returns {Promise} newly created pr object
+ */
+async function createReleasePR (body) {
+  var pr = await octokit.pulls.create({
+    owner,
+    repo,
+    title: RELEASE_PR_TITLE,
+    head: getRcBranchName(),
+    base: BASE_BRANCH,
+    body
+  });
+
+  return pr.data;
+}
+/**
+ * Get Release Candidate Branch Name
+ *
+ * @returns {string} branch name
+ */
+function getRcBranchName () {
+  return RELEASE_VERSION + '-rc';
+}
+
+/**
+ * Get Release Pull Request body content
+ *
+ * @param {Array} mergedPRs list of merged prs
+ * @returns {string} body content
+ */
+function getReleasePRBody (mergedPRs) {
+  const date = dayjs().format('DD MMMM, YYYY');
+  var body = `## Release Update - ${date}\n\n### Changelog\n`;
+
+  _.each(mergedPRs, function (pr) {
+    body += `\n* ${pr.title} #${pr.number} - @${pr.user}`;
+  });
+
+  return body;
+}
+
+/**
+ * Get Owner Name of Current Repository
+ *
+ * @returns {string} owner name
+ */
+function getOwner () {
+  const payload = github.context.payload;
+
+  return payload.repository.owner.login;
+}
+
+/**
+ * Get Current Repository Name
+ *
+ * @returns {string} repository name
+ */
+function getRepoName () {
+  const payload = github.context.payload;
+
+  return payload.repository.name;
+}
+
+/**
+ * Get Last Release object as a promise
+ *
+ * @returns {Promise} last release object
+ */
+async function getLastRelease () {
+  try {
+    var release = await octokit.repos.getLatestRelease({
+      owner: owner,
+      repo: repo
+    });
+
+    return release.data;
+  } catch (e) {
+    return;
```

Comment:
The function returns a release object. Anyone using this function rather get an error thrown instead of undefined. We need a better error handling. If we want to be silent about it we should return an object with defaults or if it does return undefined then we need to handle the undefined path maybe by existing the process?

Reply:
It will go inside `catch` when there are no releases on the repo yet. So in that case returning `undefined` is fine.
Because above we have the code like this

```javascript
const lastRelease = await getLastRelease();
const filteredPullRequests = await getPullRequests(lastRelease)

async function getPullRequests (lastRelease) {
  let commits;

  if (lastRelease) {
    commits = await getMergeCommitsSince(lastRelease.tag_name);
  } else {
    commits = await getAllMergeCommitsSinceBeginning()
  }

  return await fetchPrsForCommits(commits);
}

```

So when its undefined, we fetch `AllMergeCommitsSinceBeginning` instead of `MergeCommitsSince(lastRelease.tag_name)`.

Does that sound good?

Reply:
Yes, that sounds ok.